src/main/java/marquez/common/models/DatasourceUrn.java

```diff
@@ -19,7 +19,22 @@ public static DatasourceUrn from(@NonNull String type, @NonNull String name) {
     return fromString(value);
   }
 
-  public static DatasourceUrn fromString(String value) {
+  public static DatasourceUrn from(
+      @NonNull ConnectionUrl connectionUrl, @NonNull DatasourceName name) {
+    return from(connectionUrl.getDatasourceType().toString(), name.getValue());
+  }
+
+  public static DatasourceUrn fromString(@NonNull final String value) {
     return new DatasourceUrn(value);
   }
+
+  public DatasourceType getDatasourceType() {
+    final String datasourceType = value.split(":")[2];
+    return DatasourceType.fromString(datasourceType);
+  }
+
+  public DatasourceName getDatasourceName() {
+    final String datasourceName = value.split(":")[3];
+    return DatasourceName.fromString(datasourceName);
+  }
```

Comment:
I don't think we want URNs to be bidirectional (build <-> parse components). That is, we should only provide ways to create a URN, using it as an identifier for lookups when interfacing with our DB. @sshah-wework let me know if I'm thinking about this correctly.

Reply:
Correct, it's just an ID. We should never parse it because the details are hidden from the rest of the application intentionally.

Reply:
Makes sense - removing 👍